File: Server_Root/Server/RoomServer/GameEvent/GameEvents/EntityCreate.cpp
#include "stdafx.h"
#include "EntityCreate.h"
#include "../../../FBSFiles/EntityCreate_generated.h"

namespace GameEvent
{
	EntityCreate::EntityCreate()
	{
	}

	EntityCreate::~EntityCreate()
	{
	}

	FBS::GameEventType EntityCreate::GetType()
	{
		return FBS::GameEventType::GameEventType_EntityCreate;
	}

	const char* EntityCreate::Serialize(int* pLength)
	{
		FBS::Data::Vector3 position(m_vec3Position.x(), m_vec3Position.y(), m_vec3Position.z());
		FBS::Data::Vector3 rotation(m_vec3Rotation.x(), m_vec3Rotation.y(), m_vec3Rotation.z());

		FBS::GameEvent::EntityCreateBuilder data_builder(m_Builder);
		data_builder.add_EntityID(m_nEntityID);
		data_builder.add_MasterDataID(m_nMasterDataID);
		data_builder.add_Type(m_EntityType);
		data_builder.add_Pos(&position);
		data_builder.add_Rot(&rotation);
		auto data = data_builder.Finish();

		m_Builder.Finish(data);

		*pLength = m_Builder.GetSize();

		return (char*)m_Builder.GetBufferPointer();
	}
}

File: Client_Root/Client/Assets/Scripts/GameEvent/GameEvents/EntityCreate.cs
using UnityEngine;
using FlatBuffers;

namespace GameEvent
{
    public class EntityCreate : IGameEvent
    {
        public int m_nEntityID;
        public int m_nMasterDataID;
        public FBS.Data.EntityType m_EntityType;
        public Vector3 m_vec3Position;
        public Vector3 m_vec3Rotation;

        public override FBS.GameEventType GetEventType()
        {
            return FBS.GameEventType.EntityCreate;
        }

        public override bool Deserialize(byte[] bytes)
        {
            var buf = new ByteBuffer(bytes);

            var data = FBS.GameEvent.EntityCreate.GetRootAsEntityCreate(buf);

            m_nEntityID = data.EntityID;
            m_nMasterDataID = data.MasterDataID;
            m_EntityType = data.Type;
            m_vec3Position = new Vector3(data.Pos.X, data.Pos.Y, data.Pos.Z);
            m_vec3Rotation = new Vector3(data.Rot.X, data.Rot.Y, data.Rot.Z);

            return true;
        }

        public override string ToString()
        {
            return string.Format("{0}, EntityID : {1}, MasterDataID : {2}, EntityType : {3}, Position : {4}, Rotation : {5}", base.ToString(), m_nEntityID, m_nMasterDataID, m_EntityType, m_vec3Position.ToString(), m_vec3Rotation.ToString());
        }
    }
}
